### PR TITLE
Fix Eagle3 prefix cache hidden state flow

### PR DESCRIPTION
**Pull request by yhl-amd**
## Summary
- Keep Eagle3 draft MHA block tables aligned with target metadata during prefix-cache decode.
- Move Eagle3 draft normalization back into `compute_logits` so the draft loop passes prenorm hidden states between steps.

## Test plan
- ReadLints on `atom/models/eagle3_llama.py` and `atom/spec_decode/eagle.py`.
- Python syntax compile for the same files.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: atom/spec_decode/eagle.py [hunk 1]
```diff
@@ -378,19 +377,11 @@ def propose(
...
-                if draft_uses_mha:
-                    ret_hidden_states, ret_hidden_prenorm = model_output
-                else:
-                    ret_hidden_states = model_output
-                    ret_hidden_prenorm = None
 
                 sample_hidden_states = (
-                    torch.index_select(ret_hidden_states, 0, last_token_indices)
+                    torch.index_select(model_output, 0, last_token_indices)
                     if i == 0
```
Comment (valarLip) [inline atom/spec_decode/eagle.py:383]: keep the name ret_hidden_states, and remove "if i ==0"...

**Review by valarLip — CHANGES_REQUESTED** [review]

**Review by valarLip — APPROVED** [review]
